marklogic-client-api/src/main/java/com/marklogic/client/impl/DocumentWriteOperationImpl.java

```diff
@@ -70,4 +81,37 @@ public DocumentMetadataWriteHandle getMetadata() {
   public AbstractWriteHandle getContent() {
     return content;
   }
+
+  @Override
+  public int compareTo(DocumentWriteOperation o) {
+    if(o == null)
+      throw new NullPointerException("DocumentWriteOperation cannot be null");
+
+    if(this.getUri() instanceof String && o.getUri() instanceof String)
+      return getUri().compareTo(o.getUri());
+
+    if(this.getUri() == null && o.getUri() instanceof String)
+      return -1;
+
+    if(this.getUri() instanceof String && o.getUri()==null)
+      return 1;
+
+    if(this.getUri() == null && o.getUri() == null)
+      super.hashCode();
```

Comment:
compareTo() is defined to return -1, 0, or 1.  

super.hashCode() will rarely be one of these 3 numbers.

Would super.compareTo(o) provide a way to fall back to the default object collation?